MemoryMeter.c

```diff
@@ -73,37 +77,15 @@ static void MemoryMeter_display(const Object* cast, RichString* out) {
    Meter_humanUnit(buffer, this->total, sizeof(buffer));
    RichString_appendAscii(out, CRT_colors[METER_VALUE], buffer);
 
-   Meter_humanUnit(buffer, this->values[MEMORY_METER_USED], sizeof(buffer));
-   RichString_appendAscii(out, CRT_colors[METER_TEXT], " used:");
-   RichString_appendAscii(out, CRT_colors[MEMORY_USED], buffer);
-
-   /* shared memory is not supported on all platforms */
-   if (isNonnegative(this->values[MEMORY_METER_SHARED])) {
-      Meter_humanUnit(buffer, this->values[MEMORY_METER_SHARED], sizeof(buffer));
-      RichString_appendAscii(out, CRT_colors[METER_TEXT], " shared:");
-      RichString_appendAscii(out, CRT_colors[MEMORY_SHARED], buffer);
-   }
-
-   /* compressed memory is not supported on all platforms */
-   if (isNonnegative(this->values[MEMORY_METER_COMPRESSED])) {
-      Meter_humanUnit(buffer, this->values[MEMORY_METER_COMPRESSED], sizeof(buffer));
-      RichString_appendAscii(out, CRT_colors[METER_TEXT], " compressed:");
-      RichString_appendAscii(out, CRT_colors[MEMORY_COMPRESSED], buffer);
-   }
-
-   Meter_humanUnit(buffer, this->values[MEMORY_METER_BUFFERS], sizeof(buffer));
-   RichString_appendAscii(out, settings->showCachedMemory ? CRT_colors[METER_TEXT] : CRT_colors[METER_SHADOW], " buffers:");
-   RichString_appendAscii(out, settings->showCachedMemory ? CRT_colors[MEMORY_BUFFERS_TEXT] : CRT_colors[METER_SHADOW], buffer);
-
-   Meter_humanUnit(buffer, this->values[MEMORY_METER_CACHE], sizeof(buffer));
-   RichString_appendAscii(out, settings->showCachedMemory ? CRT_colors[METER_TEXT] : CRT_colors[METER_SHADOW], " cache:");
-   RichString_appendAscii(out, settings->showCachedMemory ? CRT_colors[MEMORY_CACHE] : CRT_colors[METER_SHADOW], buffer);
-
-   /* available memory is not supported on all platforms */
-   if (isNonnegative(this->values[MEMORY_METER_AVAILABLE])) {
-      Meter_humanUnit(buffer, this->values[MEMORY_METER_AVAILABLE], sizeof(buffer));
-      RichString_appendAscii(out, CRT_colors[METER_TEXT], " available:");
-      RichString_appendAscii(out, CRT_colors[METER_VALUE], buffer);
+   /* print the memory classes in the order supplied (specific to each platform) */
+   for (unsigned int memoryClassIdx = 0; memoryClassIdx < Platform_numberOfMemoryClasses; memoryClassIdx++) {
+      if (!settings->showCachedMemory && Platform_memoryClasses[memoryClassIdx].countsAsCache)
+         continue; // skip reclaimable cache memory classes if "show cached memory" is not ticked
```

Comment:
Some visual space doesn't hurt ^^
```suggestion
      if (!settings->showCachedMemory && Platform_memoryClasses[memoryClassIdx].countsAsCache)
         continue; // skip reclaimable cache memory classes if "show cached memory" is not ticked

```